### ISSUE TITLE
[QUESTION] After the activity is completed and worker is terminated (graceful), the last event is still ActivityTaskScheduled

### ISSUE THREAD

**Issue by khoerintus** [spiral/roadrunner#648]
Hi @48d90782 , thanks for the fix.

But I encountered another issue.
When the activity is complete and the worker is terminated, but in temporal web, the last event in the workflow is still ActivityTaskScheduled.
I expected when the activity is complete and worker is successfully terminated, I can see the last event in temporal web is ActivityTaskCompleted.

I already update my samples code [here](https://github.com/khoerintus/samples-php), if you need it.
I just lower the sleep to 20s.

_Originally posted by @khoerintus in https://github.com/spiral/roadrunner/issues/610#issuecomment-827694080_

**Comment by 48d90782** [comment]
@khoerintus Hey. Could you please try to update docker-compose components to the latest verions:
1. `temporalio/web:1.9.0`
2. `temporalio/admin-tools:1.8.2`
3. `temporalio/auto-setup:1.8.2`

**Comment by khoerintus** [comment]
Hi @48d90782 , I already update the temporal to latest version and retry it. But the result is still same as above.
Here is the screenshots:
The Activity task is completed from worker, and worker is terminated:
![Screenshot from 2021-05-12 15-41-53](https://user-images.githubusercontent.com/9275991/117946916-ea479680-b339-11eb-96e9-87d27745f9e5.png)

But in temporal web, the workflow is still running, and the last activity task is still pending.
![Screenshot from 2021-05-12 15-43-25](https://user-images.githubusercontent.com/9275991/117947084-0fd4a000-b33a-11eb-9f95-b239c9667243.png)

**Comment by 48d90782** [comment]
Hey @khoerintus. I see that the activity Plugin isn't stopped properly. I guess this is why you see such workflow status as on the screenshot, but I'll double-check that.

**Comment by 48d90782** [comment]
Ok, I dig into your problem and understand what did you mean. 😄 
This is not a bug but the actual behavior of the Temporal. Your activity (state) is stored in the temporal server. When you stop the worker, it does not mean that RR will stop the activity immediately. It will be trying to continue execution due to the timeout period (5min by default). Only after that period, if the worker is still down, the Temporal will stop the activity (and the reason will be - `timeout`). You may, for example start RR, fire the `LongRunningActivity.php`, then stop RR and stop `LongRunningActivity.php`. Then without starting `LongRunningActivity.php`, start the RR server and you'll see, that the Temporal will start the counter even without re-running the activity. To cancel the activity, you should directly cancel it with a proper command https://docs.temporal.io/docs/concepts/activities/#cancellation.`

**Comment by 48d90782** [comment]
Forgot - extended status:
![image](https://user-images.githubusercontent.com/8040338/118013651-027bdd80-b35b-11eb-8d3b-503e11c3f7a3.png)

**Comment by khoerintus** [comment]
Hi @48d90782 ,
I am sorry, but I think that we don't understand each other.
For you to understand easier, here is the detail action and event that actually happened and what I expect:

00:00:00  -  start roadrunner worker with grace_period 30s
00:00:00  -  start an activity which sleep for 20s
00:00:01  -  terminate roadrunner worker, start graceful shutdown, waiting for 30s
00:00:20  -  activity completed, but does not send completed event to temporal server (I expect that when this activity completed, activity state in temporal will update to completed)
00:00:21  -  worker stop gracefully

**Comment by 48d90782** [comment]
Hey @khoerintus, let me double-check this and re-open the issue if a completed event does not arrive in the temporal server.

**Comment by rustatian** [comment]
Ok guys, got u, reopening the issue. I guess, that I missed something. I'll investigate this, thanks.

**Comment by rustatian** [comment]
@khoerintus , @changwuf31 Hey, I guess this is normal behavior when the worker is in the stopping state according to the temporal `go-sdk` sources: 
https://github.com/temporalio/sdk-go/blob/6d79cfde669a309a23463318d7b5fe45a9dfd751/internal/internal_task_pollers.go#L879.
I can also see this error in the logs: `INFO    temporal        Task processing failed with error       {"Namespace": "default", "TaskQueue": "default", "WorkerID": "197549@rustatian@default@3", "WorkerType": "ActivityWorker", "Error": "worker stopping"}`.
When the worker is in the `stopping` state it returns the `worker stopping` error and stop all task processing (polling) for the activities and workflows.
It's hard to say if that's behavior correct or not. But I generally agree, that if the worker is in the `stopping` state, you can skip checking for the activity completion state since the worker was interrupted.
You may also ask this question at the Temporal support forum: https://community.temporal.io/

**Comment by rustatian** [comment]
ref?: https://github.com/temporalio/sdk-go/issues/480

**Comment by rustatian** [comment]
@khoerintus We are waiting for the temporal team to fix this issue, then I'll update dependencies.
